bn-apps/cordapp-updates-distribution/corda-updates-transport/src/main/kotlin/net/corda/cordaupdates/transport/flows/PeekResourceFlow.kt

```diff
@@ -0,0 +1,63 @@
+package net.corda.cordaupdates.transport.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.cordaupdates.transport.flows.Utils.toCordaException
+import net.corda.cordaupdates.transport.flows.Utils.transporterFactory
+import net.corda.core.flows.FlowLogic
+import net.corda.core.flows.FlowSession
+import net.corda.core.flows.InitiatedBy
+import net.corda.core.flows.InitiatingFlow
+import net.corda.core.flows.StartableByRPC
+import net.corda.core.flows.StartableByService
+import net.corda.core.identity.CordaX500Name
+import net.corda.core.utilities.unwrap
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils
+import org.eclipse.aether.repository.RemoteRepository
+import org.eclipse.aether.spi.connector.transport.PeekTask
+import java.lang.Exception
+import java.net.URI
+
+/**
+ * Peeks resource in a remote repository. Used internally by Maven Resolver.
+ *
+ * @param resourceLocation location of the resource provided by Maven Resolver
+ * @param repoHosterName x500Name of the repository hoster
+ */
+@InitiatingFlow
+@StartableByRPC
+@StartableByService
+class PeekResourceFlow(private val resourceLocation : String, private val repoHosterName : String) : FlowLogic<Unit>() {
+    @Suspendable
+    override fun call() {
+        val repoHosterParty = serviceHub.identityService.wellKnownPartyFromX500Name(CordaX500Name.parse(repoHosterName))!!
```

Comment:
Ditto - don't NPE.

Perhaps this pattern of "try to match and throw a helpful exception if we miss" should be in the platform ... wanna write a PR to add it? You can always copy/paste the resulting code into a BNO-app specific function afterwards.

Reply:
You mean to throw an exception from `wellKnownPartyFromX500Name` if a party wasn't found instead of returning a nullable type? But that would break an API wouldn't it?

Reply:
No I mean adding a new API that does that, but with the helpful error message that suggests what you probably meant in the exception? Like:

```
class UnknownPartyException(val request: String, val suggestions: List<Party>) : FlowException

class IdentityService {   // or wherever
    /** @throws UnknownPartyException if there was no match */
    fun partyFromName(name: String): Party {
        val suggestions = partiesFromName(name)
        return suggestions.singleOrNull() ?: throw UnknownPartyException(name, suggestions)
    }
}
```

or words to that effect

Reply:
Yep, will add that